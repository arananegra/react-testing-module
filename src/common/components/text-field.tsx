import * as React from 'react';
import MaterialTextField, { TextFieldProps as MaterialTextFieldProps } from '@material-ui/core/TextField';
//Removed due to tests conflicts
//import { Overwrite } from '@material-ui/types';

export interface TextFieldProps {
  name: string;
  label: string;
  value?: string;
  onChange: (field: string, value, event?) => void;
  error?: string;
  type?: string;
  onBlur?: (field: string, value, event?) => void;
  className?: string;
}

export const TextField: React.FC<TextFieldProps> = props => {
  const {
    name,
    label,
    onChange,
    value,
    error,
    type,
    onBlur,
    className,
    ...otherProps
  } = props;
  const hasError = Boolean(error);

  const handleChange = e => {
    onChange(name, e.target.value, e);
  };

  const handleBlur = e => {
    if (onBlur) {
      onBlur(name, e.target.value, e);
    }
  };

  return (
    <MaterialTextField
      {...otherProps}
      className={className}
      label={label}
      margin="normal"
      value={value}
      type={type}
      onChange={handleChange}
      inputProps={
        { ['data-testid']: 'mui-textfield' }
      }
      onBlur={handleBlur}
      error={hasError}
      helperText={hasError && error}
    />
  );
};

TextField.defaultProps = {
  type: 'text',
};
