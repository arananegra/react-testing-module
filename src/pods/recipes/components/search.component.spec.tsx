import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { SearchComponent } from './search.component';
import { TextField } from 'common/components/text-field';

describe('Search component tests', () => {
  it('should render a happy search component', () => {
    // Arrange
    const props = {
      onSearch: jest.fn(),
    };

    // Act
    const { getByTestId, asFragment } = render(<SearchComponent {...props} />);
    const searchComponent = getByTestId('search-component') as HTMLInputElement;
    //Assert
    expect(searchComponent).not.toBeUndefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onSearch when event is fired', () => {
    // Arrange
    const externalProps = {
      onSearch: jest.fn(),
    };
    // Act
    const { getByTestId } = render(<SearchComponent {...externalProps} />);
    const textFieldElement = getByTestId('mui-textfield') as HTMLInputElement;
    fireEvent.change(textFieldElement, { target: { value: 'John' } });
    // Assert
    expect(externalProps.onSearch).toHaveBeenCalled();
  });
});
