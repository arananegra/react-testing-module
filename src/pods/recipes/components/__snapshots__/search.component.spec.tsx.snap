// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component tests should render a happy search component 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-textField-1 MuiFormControl-marginNormal"
    data-testid="search-component"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink="false"
    >
      Search
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        data-testid="mui-textfield"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
